src/service.js

```diff
@@ -182,12 +183,23 @@ class Service {
       data[this.id] = id;
     }
 
-    return this.Model
-      .update(params.query, { $set: data }, options)
-      .lean(this.lean)
-      .exec()
-      .then(() => this._getOrFind(id, params))
-      .catch(errorHandler);
+    let promise;
+
+    // We need this shitty hack because update doesn't return
+    // a promise properly when runValidators is true. WTF!
+    try {
+      promise = this.Model
```

Comment:
Looking at the code again I don't think we really need to switch to an internal `_update` method. They are different enough in the way that they use Mongoose methods and that's a refactor that's not really necessary and I don't have time for at the moment. Something we can come back to in the future if necessary. It shouldn't hold up getting these fixes in.